src/libs/OnboardingUtils.ts

```diff
@@ -12,8 +12,22 @@ const supportedIntegrationsInNewDot = ['quickbooksOnline', 'quickbooksDesktop',
  * @returns boolean - True if user should be redirected to old dot
  */
 function shouldOnboardingRedirectToOldDot(companySize: OnboardingCompanySize | undefined, userReportedIntegration: OnboardingAccounting | undefined): boolean {
+    // Desktop users should never be redirected to old dot
+    if (getPlatform() === CONST.PLATFORM.DESKTOP) {
+        return false;
+    }
+
+    // Check if the integration is supported in NewDot
     const isSupportedIntegration = (!!userReportedIntegration && supportedIntegrationsInNewDot.includes(userReportedIntegration)) || userReportedIntegration === null;
-    return getPlatform() !== CONST.PLATFORM.DESKTOP && (!isSupportedIntegration || companySize !== CONST.ONBOARDING_COMPANY_SIZE.MICRO);
+
+    // Don't redirect if integration is supported and company size is MICRO or SMALL
+    const isMicroOrSmallCompany = companySize === CONST.ONBOARDING_COMPANY_SIZE.MICRO || companySize === CONST.ONBOARDING_COMPANY_SIZE.SMALL;
```

Comment:
The only real change in this PR is `|| companySize === CONST.ONBOARDING_COMPANY_SIZE.SMALL`.
Every other change here is to make this block more readable because at least for me the previous code with a lot of negation and lack of early returns made it hard for me to read.